tests/ui/type-alias-impl-trait/ice-failed-to-resolve-instance-for-110696.rs

```diff
@@ -42,11 +42,11 @@ impl<T: MyFrom<Phantom2<DummyT<U>>>, U> MyIndex<DummyT<T>> for Scope<U> {
     //~^ ERROR the type parameter `T` is not constrained by the impl
     type O = T;
     fn my_index(self) -> Self::O {
-        //~^ ERROR item does not constrain
         MyFrom::my_from(self.0).ok().unwrap()
     }
 }
 
 fn main() {
     let _pos: Phantom1<DummyT<()>> = Scope::new().my_index();
+    //~^ ERROR: no method named `my_index` found for struct `Scope`
```

Comment:
This one is funny and should probably have some more taint tracking to get eliminated (same with the TAIT cycle above)